src/controllers/hotels.controller.js

```diff
@@ -100,11 +102,72 @@ class Hotel {
       type,
       description,
       cost,
-      status
+      status,
     });
 
     return Responses.handleSuccess(201, 'Room added successfully', res, room);
   }
+
+  /**
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} responses
+   */
+  async getMostVisitedDestination(req, res) {
+    const page = (req.query.page * 1) || 1;
+    const results = [];
+    let trips = [];
+    const requests = await getAllRequestsHotels();
+
+    if (!requests.length) {
+      return Responses.handleError(404, 'There is no past travel yet.', res);
+    }
+
+    const requestTrips = requests.map(el => el.trips);
+
+    for (let i = 0; i < requestTrips.length; i += 1) {
+      trips = trips.concat(requestTrips[i]);
+    }
+
+    const counter = {};
+    // let maxCount = 1;
+
+    for (let i = 0; i < trips.length; i += 1) {
+      const el = trips[i].hotel.id;
+
+      if (counter[el] === undefined) {
+        counter[el] = 1;
+      } else {
+        counter[el] += 1;
+      }
+    }
+
+    const sortable = [];
+
+    // eslint-disable-next-line guard-for-in,no-restricted-syntax
+    for (const element in counter) {
+      sortable.push([element, counter[element]]);
+    }
+    sortable.sort((a, b) => a[1] - b[1]);
+    const five = sortable.slice(page - 1, page * 5);
+    const fiveKeys = five.map(item => item[0]);
+
+    const fivePaginated = await hotelService.mostTravelled(fiveKeys);
+
+    for (let i = 0; i < five.length; i += 1) {
+      const hotel = fivePaginated.find(item => `${item.id}` === five[i][0]);
+      results.unshift({
+        count: five[i][1],
+        hotel,
+      });
+    }
```

Comment:
Great work, @gildniy, However, these code are really complex 
I think you can try this SQL query to get most visited hotels
```
select count(trips."hotelId") as visits, hotels."name", hotels."id" as hotelId from hotels
Left Join trips
On trips."hotelId" = hotels."id"
Group By hotels."id", hotels."name"
Order By visits DESC
```
![image](https://user-images.githubusercontent.com/15703806/70513500-726caf80-1b3a-11ea-9911-180600fad089.png)